### PR TITLE
src detector line number fix

### PR DESCRIPTION
**Pull request by vipati**
- Fixed logic to correctly show the line number of the code where error/warning is present
- It will help with disable-next-line rule

### CODE REVIEW

File: packages/rules-typescript/src/src-detector.ts [hunk 1]
```diff
@@ -50,11 +50,17 @@ export class SrcDetector implements PackageRule {
 
   getImportPaths(sourceFile: SourceFile): ImportPathAndLineNumber[] {
     const importPaths: ImportPathAndLineNumber[] = [];
+    let lineNumber = 1;
     sourceFile.forEachChild(n => {
+      const totalLines = n.getFullText().split(/\r?\n/).length;
+      lineNumber = lineNumber + totalLines - 1;
       if (isImportDeclaration(n)) {
+        console.log("==================================>>>>>>>>>>>>");
+        console.log("path : " + n.moduleSpecifier.getText());
```
Comment (jofrey-a-msft-production) [inline packages/rules-typescript/src/src-detector.ts:59]: Did you mean to remove these console logs?

Comment (vipati) [reply packages/rules-typescript/src/src-detector.ts]: Yes. I removed it.

File: packages/rules-typescript/src/tests/src-detector.test.ts [hunk 2]
```diff
@@ -23,3 +24,17 @@ test("Should fail if `src` detected in imports", async () => {
   const result = sut.checkImportPaths(asBollFile("a"), importPaths);
   assert.strictEqual(ResultStatus.failure, result[0].status);
 });
+
+test("Should fail if references to node_modules exist in source code", async () => {
```
Comment (jofrey-a-msft-production) [inline packages/rules-typescript/src/tests/src-detector.test.ts:28]: Should this say: "Should fail if references to **src** exist in source code"?

**Review by jofrey-a-msft-production — APPROVED** [review]

File: packages/rules-typescript/src/tests/src-detector.test.ts (hunk 2 above)
Comment (vipati) [reply packages/rules-typescript/src/tests/src-detector.test.ts]: Updated

**Pull request merged**